### PR TITLE
Adds payout module for safe interaction

### PR DESCRIPTION
**Pull request by 3commascapital**
* not hooked up yet, only module
* mnemonic used for now - can easily switch to privkey

### CODE REVIEW

**Comment by pavlovcik** [comment]
How is accounting handled? Also we should do private key instead of mnemonic.

File: .env.example [hunk 1]
```diff
@@ -7,3 +7,7 @@ LOG_LEVEL=debug
 
 # Go to https://smee.io/new set this to the URL that you are redirected to.
 WEBHOOK_PROXY_URL=
+
```
Comment (zgorizzo69) [inline .env.example:10]: I think we miss the `MNEMONIC` env

File: src/utils/payout.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+export const providerKeyByChainId = new Map<number, string>([
+  [1, process.env.RPC1_ALCHEMY_KEY as string],
+  [5, process.env.RPC5_ALCHEMY_KEY as string], // goerli - safe compatable
+])
+
+export const payout = async (args: PayoutParams) => {
+  // collect baseline info
+  const tokenAddress = ethers.utils.getAddress(args.token)
```
Comment (zgorizzo69) [inline src/utils/payout.ts:25]: shouldn't we make sure the token address is whitelisted ?

File: src/utils/payout.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const isEth = tokenAddress === ethers.constants.AddressZero
+  const destinationAddress = ethers.utils.getAddress(args.to)
+  const mnemonic = process.env.MNEMONIC as string
+  const network = args.network || 1
```
Comment (zgorizzo69) [inline src/utils/payout.ts:29]: we should also check that we allow the network and that we have the corresponding alchemy key

**Review by zgorizzo69 — CHANGES_REQUESTED** [review]

File: src/utils/payout.ts (hunk 2 above)
Comment (pavlovcik) [reply src/utils/payout.ts:25]: That's a nice detail. This is to make sure we don't waste gas I presume?

File: src/utils/payout.ts (hunk 3 above)
Comment (pavlovcik) [reply src/utils/payout.ts:29]: Also nice detail.

File: .env.example (hunk 1 above)
Comment (pavlovcik) [reply .env.example:10]: We will use PRIVATE_KEY or something of the like, instead of mnemonic.

**Comment by 0xcodercrane** [comment]
@3commascapital  would you rebase from main and continue working? we had a bunch of refactors

File: src/utils/payout.ts (hunk 2 above)
Comment (3commascapital) [reply src/utils/payout.ts:25]: correct. good addition @zgorizzo69

Comment (3commascapital) [reply src/utils/payout.ts:25]: do you mean something specific @zgorizzo69 when you use whitelisted? i was just thinking - balance, and allowed is set to cover the amount. is there anything else?

**Comment by pavlovcik** [comment]
Screenshot 2022-12-02 at 15 37 42

**Pull request closed without merging**